enhancements/ingress/lb-allowed-source-ranges.md

```diff
@@ -0,0 +1,261 @@
+`.Spec.LoadBalancerSourceRanges` field of the LoadBalancer Service in a previous version. 
+This risk is mitigated by marking the cluster as non-upgradable in such case.
+
+### Drawbacks
+
+This enhancement may bring additional complexity for the customers who are 
+used to modify directly the `Spec.LoadBalancerSourceRanges` of the LoadBalancer 
+Service for the same purposes. 
+
+## Design Details
+
+### Open Questions [optional]
+
+1. On Kubernetes API reference, the field 
+[`loadBalancerSourceRanges`](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.24/#service-v1-core) 
+has the warning "This field will be ignored if the cloud-provider does not support the feature.".
+However, there is no information about which platforms support this. Is there any platform
+OpenShift runs on which does not support this feature?
+
+A. I could only find this statement from Kubernetes
+[v1.18 website repository](https://github.com/kubernetes/website/blob/dev-1.18/content/en/docs/tasks/access-application-cluster/configure-cloud-provider-firewall.md): 
+"This feature is currently supported on Google Compute Engine, Google Kubernetes Engine, 
+AWS Elastic Kubernetes Service, Azure Kubernetes Service, and IBM Cloud Kubernetes Service". 
+The current version of the document does not mention `.Spec.LoadBalancerSourceRanges` anymore.
+
+### Test Plan
+
+The operator that manages the IngressController Deployment and related
+resources has unit test coverage; for this enhancement, the unit tests are
+expanded to cover the additional functionality.  
+
+The operator has end-to-end tests; for this enhancement, a test is added that verifies
+the following scenarios:
+
+1. Adding/deleting/changing `AllowedSourceRanges` on the IngressController and observing `.Spec.LoadBalancerSourceRanges` on the LoadBalancer Service.
+
+2. Adding `service.beta.kubernetes.io/load-balancer-source-ranges` annotation to a LoadBalancer Service and observing cluster's upgradable status.
+
+There is an upgrade test in openshift/origin to make sure that the migration workflow explained under **Variation** above works seamlessly.
+
+### Graduation Criteria
+
+N/A
+
+#### Dev Preview -> Tech Preview
+
+N/A. This feature will go directly to GA.
+
+#### Tech Preview -> GA
+
+N/A. This feature will go directly to GA.
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+On upgrade, `AllowedSourceRanges` is omitted by default, which is consistent with the
+feature's absence in older versions.  On downgrade, the IngressController
+Deployment is updated, and `AllowedSourceRanges` is automatically
+removed from the Deployment.
+
+The cluster will be marked as non-upgradable:
+- On OpenShift 4.11, if `Spec.LoadBalancerSourceRanges` of the LoadBalancer Service is set.
+- On OpenShift 4.12, if `service.beta.kubernetes.io/load-balancer-source-ranges` annotation for the LoadBalancer
+  Service is set.
```

Comment:
This is useful information.  However, it describes what the operator does without describing why.  A few words of elaboration would help, something like the following: "The intention is to ensure that the cluster admin does not have configuration in `Spec.LoadBalancerSourceRanges` prior to upgrading to OpenShift 4.12, since the upgraded operator would overwrite that field.  Then once the cluster is upgraded to OpenShift 4.12, the intention is to guide the cluster admin towards using the IngressController API and deprecate use of the service annotation for ingress."

Reply:
Thanks, I've added this to the EP.